common/network-common/src/main/java/org/apache/spark/network/buffer/FileSegmentManagedBuffer.java

```diff
@@ -17,13 +17,12 @@
 
 package org.apache.spark.network.buffer;
 
-import java.io.File;
-import java.io.FileInputStream;
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.RandomAccessFile;
+import java.io.*;
 import java.nio.ByteBuffer;
 import java.nio.channels.FileChannel;
+import java.util.zip.Adler32;
```

Comment:
would be better to abstract out the checksum functions into some class so it's easier to change in the future.